fix(linter): print total rule # when using a single nested config

connorshea opened the pull request:
For a while, we've been skipping the printing of rule runs unless `--disable-nested-config` was passed. This was a result of the nested_configs value including the base config (see #16356). When we were checking whether that value was empty, it would always have at least 1 item (unless nested configs were entirely disabled).

As part of this, I've updated some configs in the fixtures to disable correctness rules, otherwise we will end up with a bunch of fixture snapshots needing updates every time a new default rule is added.

It's probably easiest to review this one commit at a time.

camc314 commented on crates/oxc_linter/src/config/config_store.rs:
```diff
@@ -1018,34 +1019,29 @@ mod test {
+        // TODO: Can remove this addition of the base config when we fix https://github.com/oxc-project/oxc/issues/16356
+        nested_configs.insert(PathBuf::new(), base.clone());
+        // Then add another so we have more than just the base config here.
+        nested_configs.insert(PathBuf::from("nested"), base);
```
oh hah i thought this was real code (not test code) and it confused me so much

camc314 commented on crates/oxc_linter/src/config/config_store.rs:
```diff
@@ -297,7 +297,8 @@ impl ConfigStore {
     /// Returns the number of rules, optionally filtering out tsgolint rules if type_aware_enabled is false.
     pub fn number_of_rules(&self, type_aware_enabled: bool) -> Option<usize> {
-        if !self.nested_configs.is_empty() {
+        // This includes the base config for now, so 1 is fine :)
```
```suggestion
        // This includes the base config for now, so 1 is fine :) Reference: https://github.com/oxc-project/oxc/issues/16356
```

connorshea committed:
fix(linter): print total rule # when using a single nested config (#17517)

For a while, we've been skipping the printing of rule runs unless `--disable-nested-config` was passed. This was a result of the nested_configs value including the base config (see #16356). When we were checking whether that value was empty, it would always have at least 1 item (unless nested configs were entirely disabled).

As part of this, I've updated some configs in the fixtures to disable correctness rules, otherwise we will end up with a bunch of fixture snapshots needing updates every time a new default rule is added.

It's probably easiest to review this one commit at a time.
